lib/vrl/compiler/src/expression/if_statement.rs

```diff
@@ -17,11 +17,9 @@ pub struct IfStatement {
 
 impl IfStatement {
     pub(crate) fn noop() -> Self {
-        let literal = Literal::Boolean(false);
-        let predicate = Predicate::new_unchecked(vec![Expr::Literal(literal)]);
+        let predicate = Predicate::new_unchecked(vec![]);
```

Comment:
Predicate's `Expression` implementation falls back to `Value::Boolean(false)` if the vec is empty. This can't happen in regular programs, as this calls `new_unchecked`, whereas regular `new` validates the expressions in the if-statement predicate.